ee/tabby-webserver/src/service/event_logger.rs

```diff
@@ -23,14 +23,27 @@ pub fn create_event_logger(db: DbConn) -> impl EventLogger + 'static {
 
 impl EventLogger for DbEventLogger {
     fn write(&self, x: LogEntry) {
+        let Ok(event_json) = serde_json::to_string_pretty(&x.event) else {
+            warn!("Failed to convert event {x:?} to JSON");
+            return;
+        };
         match x.event {
             Event::View { completion_id, .. } => {
                 let db = self.db.clone();
                 tokio::spawn(async move {
                     log_err(
                         db.add_to_user_completion(x.ts, &completion_id, 1, 0, 0)
                             .await,
-                    )
+                    );
+                    log_err(
+                        db.create_user_event_lookup_user(
+                            completion_id,
+                            "view".into(),
```

Comment:
define `DbEnum` for `Event`.

Reply:
These are always strings at the db layer, the `DbEnum` is used at the service layer

Reply:
Exactly - thus we should define DbEbum for Event and invoke as_db_str here?